docs/_static/link_gen/link.js

```diff
@@ -117,6 +117,18 @@ function changeTab(div) {
     }
 }
 
+/**
+ * Return name of directory git will clone given repo to.
+ *
+ * nbgitpuller needs to redirect users to *inside* the directory it
+ * just cloned. We copy the logic git itself uses to determine that.
+ * See https://github.com/git/git/blob/1c52ecf4ba0f4f7af72775695fee653f50737c71/builtin/clone.c#L276
+ */
+function generateCloneDirectoryName(gitCloneUrl) {
+    var lastPart = gitCloneUrl.split('/').slice(-1)[0];
```

Comment:
I think I liked better using the `pop` function istead of `slice(-1)[0]`. I find it less brain consuming than thinking about indexes :sweat_smile:
(p.s: Did I got it right? they do the same thing, right t?)

Reply:
[slice](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/slice) does not modify the original array, while [pop](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/pop) does. In general, it's a good idea to not mutate things in-place unless necessary. It doesn't matter in this particular case, but I think in general `slice` is the way to go.

Reply:
Oh right! Thank you for making this clear for me.